pkg/csi/service/common/vsphereutil.go

```diff
@@ -415,14 +440,17 @@ func CreateFileVolumeUtilOld(ctx context.Context, clusterFlavor cnstypes.CnsClus
 			// If DatastoreURL is not specified in StorageClass, get all datastores
 			// from TargetvSANFileShareDatastoreURLs in vcenter configuration.
 			for _, TargetvSANFileShareDatastoreURL := range manager.VcenterConfig.TargetvSANFileShareDatastoreURLs {
-				datastoreMoref, err := getDatastore(ctx, vc, TargetvSANFileShareDatastoreURL)
+				datastoreInfoObj, err := getDatastoreInfo(ctx, vc, TargetvSANFileShareDatastoreURL)
 				if err != nil {
 					log.Errorf("failed to get datastore %s. Error: %+v", TargetvSANFileShareDatastoreURL, err)
 					// TODO: Need to figure out the fault extracted from getDatastore.
 					// Currently, just return csi.fault.Internal.
 					return "", csifault.CSIInternalFault, err
 				}
-				datastores = append(datastores, datastoreMoref)
+				if filterSuspendedDatastores && vsphere.IsVolumeCreationSuspended(ctx, datastoreInfoObj) {
+					continue
+				}
+				datastores = append(datastores, datastoreInfoObj.Datastore.Reference())
```

Comment:
You can add a check after this that if `datastores` slice is empty, return an error. Otherwise we will pass empty list to the create spec.